src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -627,15 +637,25 @@ private void ReportAssignmentOperatorError(AssignmentExpressionSyntax node, Diag
 
         private static void ReportBinaryOperatorError(ExpressionSyntax node, DiagnosticBag diagnostics, SyntaxToken operatorToken, BoundExpression left, BoundExpression right, LookupResultKind resultKind)
         {
-            if ((operatorToken.Kind() == SyntaxKind.EqualsEqualsToken || operatorToken.Kind() == SyntaxKind.ExclamationEqualsToken) &&
-                (left.IsLiteralDefault() && right.IsLiteralDefault()))
+            bool leftDefault = left.IsLiteralDefault();
+            bool rightDefault = right.IsLiteralDefault();
+            if ((operatorToken.Kind() == SyntaxKind.EqualsEqualsToken || operatorToken.Kind() == SyntaxKind.ExclamationEqualsToken))
             {
-                Error(diagnostics, ErrorCode.ERR_AmbigBinaryOpsOnDefault, node, operatorToken.Text);
-                return;
-            }
+                if (leftDefault && rightDefault)
+                {
+                    Error(diagnostics, ErrorCode.ERR_AmbigBinaryOpsOnDefault, node, operatorToken.Text);
+                    return;
+                }
 
-            if (left.IsLiteralDefault() || right.IsLiteralDefault())
+                if (leftDefault || rightDefault)
+                {
+                    Error(diagnostics, ErrorCode.ERR_BadBinaryOps, node, operatorToken.Text, (leftDefault ? right : left).Display, "System.Object");
```

Comment:
>"System.Object" [](start = 131, length = 15)

Why do we assume that this is the type of the argument, I think we should simply get Display from left and right, no need for conditional logic or hardcoded strings.